lib/dal/src/schema/variant.rs

```diff
@@ -459,7 +459,10 @@ impl SchemaVariant {
 
         // Now add the default schema variant to the list as this is a different edge kind
         let default_schema_variant = Self::get_default_for_schema(ctx, schema_id).await?;
-        schema_variants.push(default_schema_variant);
+        // We will only add the default schema variant IF there is no existing schema variant found
+        if !schema_variants.contains(&default_schema_variant) {
+            schema_variants.push(default_schema_variant);
+        }
```

Comment:
I think we can move these lines altogether since `EdgeWeightKindDiscriminants::Use` (passed into `outgoing_targets_for_edge_weight_kind_by_index`) includes both `Use { default: true }` and `Use { default: false }`.

Reply:
I'd change line `425` to be `EdgeWeightKindDiscriminants::Use` to be more clear too. Might help others that run into the issue.